src/main/java/org/javawebstack/abstractdata/AbstractObject.java

```diff
@@ -267,4 +267,26 @@ public AbstractElement clone() {
         return new AbstractObjectCollector<>(keyFunction, valueFunction);
     }
 
+    public boolean equals(Object obj, boolean strict) {
+        if (obj == null)
+            return false;
+        if (!(obj instanceof AbstractObject))
```

Comment:
This ignores the strict flag. When strict is false, it needs to try to coerce AbstractArray into AbstractObject